modules/deployment-model/src/main/java/org/mule/runtime/deployment/model/api/DeployableArtifact.java

```diff
@@ -65,4 +66,10 @@
    */
   ConnectivityTestingService getConnectivityTestingService();
 
+  /**
+   * @return the {@link MetadataService} which can resolve the metadata of the components inside
+   * the current {@link DeployableArtifact}
+   * @see MetadataService
+   */
+  MetadataService getMetadataService();
```

Comment:
isn't this mixing concerns? It's weird to me that the MetadataService gets exposed at such a high level. @pablokraan can you confirm?